config/routes.rb

```diff
@@ -1,6 +1,19 @@
 Rails.application.routes.draw do
+  devise_for :users
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
   
+  # Define root path
+    devise_scope :user do
+      authenticated :user do
+        root :to => 'users#index', as: :authenticated_root
+      end
+      unauthenticated :user do
+        root :to => 'login#index', as: :unauthenticated_root
```

Comment:
👋 Hi there!, when I try to go to the login page, I get this error:

<img width="922" alt="image" src="https://github.com/jecode93/blog-app/assets/52934128/f26b2cd3-8430-4dd7-b02e-80f0f5bdb3f6">

---

I've noticed an issue related to the placement of the view templates for your custom `LoginController`. In a standard Rails application, the framework expects the view templates to reside in a folder that matches the name of the controller. This means for your `LoginController#index` action, Rails will look for the view at `app/views/login/index.html.erb`.

Your initial setup placed the `login` folder within the `app/views/devise` directory, which is intended only for the views of Devise's built-in controllers, like `Devise::SessionsController`. Since `LoginController` is a custom controller outside of Devise's scope, the view should not be located inside the `devise` directory.

**Corrective Action:**
- Move the `login` directory from `app/views/devise/login` to `app/views/login`. This change adheres to Rails' convention over configuration principle, ensuring that Rails finds the appropriate view templates for custom controllers.
- After reorganizing the views into the correct directory, double-check that the application functions as expected without any template errors.

This adjustment will align your project's structure with Rails conventions, thereby resolving the template error and making your project's organization more intuitive for other developers.

Let me know if you have successfully moved the views or if you encounter any issues after making these changes.

Keep up the good work!

---

Best regards,

Reply:
@topeogunleye thank you for your review.

Changes completed!